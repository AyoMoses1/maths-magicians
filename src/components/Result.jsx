/* eslint-disable react/prop-types */

import React from 'react';

function Result({ total, next, operation }) {
  return (
    <div className="result" data-testid="result">
      {total}
      {' '}
      {operation}
      {' '}
      {next}
    </div>
  );
}

export default Result;
