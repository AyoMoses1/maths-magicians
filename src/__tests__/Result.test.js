import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Result from '../components/Result';
import '@testing-library/jest-dom';

it('renders correctly', () => {
  const state = { total: 2, next: '3', operation: '+' };
  const tree = renderer.create(
    <Result
      total={state.total}
      next={state.next}
      operation={state.operation}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('The Result component renders the same content passed into total, next and operation prop', () => {
  const state = { total: 2, next: '3', operation: '+' };
  render(<Result total={state.total} next={state.next} operation={state.operation} />);
  const displayedContent = screen.getByText('2 + 3');
  expect(displayedContent).toBeInTheDocument();
});
