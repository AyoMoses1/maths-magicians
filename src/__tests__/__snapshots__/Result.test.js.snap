// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="result"
  data-testid="result"
>
  2
   
  +
   
  3
</div>
`;
